### PR TITLE
Implement semantic release

### LINKED ISSUE #154: npm-published packages should follow semver
### Expected Behavior

Scratch projects published to npm should use semver version numbers >= 1.0 so that dependents can control when to accept breaking changes.

### Actual Behavior

This repository uses "prerelease" version numbers like `0.1.0-prerelease.YYYYMMDDhhmmss` which contain no information about breaking changes.

### PR DESCRIPTION
**Pull request by cwillisf**
### Proposed changes

- Resolves ENA-286
- Resolves #154 

Implement `semantic-release` and friends.

### CODE REVIEW

File: .github/workflows/commitlint.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Lint commit messages
+on: [pull_request]
+
+concurrency:
+  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.sha }}"
+
```
Comment (skripted-io) [inline .github/workflows/commitlint.yml:6]: I noticed that for [scratch-sb1-converter](https://github.com/scratchfoundation/scratch-sb1-converter) you added "cancel-in-progress: true" here. Not sure if you intentionally omitted it or not.

**Review by skripted-io — COMMENTED** [review]
Love how this cleans up the workflow. Only have a minor question re: concurrency.

File: .github/workflows/commitlint.yml (hunk 1 above)
Comment (cwillisf) [reply .github/workflows/commitlint.yml:6]: Oops! I actually didn't mean to remove that. The great thing about writing shell scripts for this kind of thing is you can make the same mistake everywhere, all at once! ;)

Comment (cwillisf) [reply .github/workflows/commitlint.yml:6]: Oh wait, this is for the commitlint job. Yeah, I guess that doesn't really matter :)

**Pull request merged**